### PR TITLE
Create module config that works for linux and android targets

### PR DESCRIPTION
**Pull request by seanavery**
## Description
Goal here is to simplify `module` cmds between Android and Linux and provide a common module entrypoint `run.sh` that works on both targets for module registry uploads.

!!! Current changes to model name are for testing uploads and will be rebased with `erh` before merging to main.

## Todo
- [x] Create new model `viam-labs:camera:rtsp`.
- [x] Add AppImg and run.sh to module tar.
- [x] Make `run.sh` entrypoint work for both Appimage/Linux and Android.
- [x] Make single `meta.json` work for both Linux Arm64/Amd64 and Android.
- [x] Make `module` bundling cmd work on both Android and Linux.

## Testing
- [x] Test new bundle works on arm64 linux (pi4).
- [x] Test bundle works in Android emulator.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,9 +61,19 @@ build-android:
 		go build -v -tags no_cgo \
 		-o $(OUTPUT) ./cmd/module/cmd.go
 
-module.tar.gz: $(OUTPUT) run.sh
-	# todo: dedup with 'make module' command
-	tar czf $@ $^ -C $(FFMPEG_PREFIX) lib
+.PHONY: module
+ifeq ($(GOOS),android)
```
Comment (seanavery) [inline Makefile]: @abe-winter There is probably a better way to do this but current workaround is simple if check on GOOS.

- If Android, copy over binary, run.sh, and ffmpeg libs
- If Linux, copy over AppImg and run.sh

File: Makefile [hunk 2]
```diff
@@ -3,8 +3,8 @@ GOARCH ?= $(shell go env GOARCH)
 ARCH ?= $(shell uname -m)
 TARGET_IP ?= 127.0.0.1
 API_LEVEL ?= 29
+MOD_VERSION ?= 0.0.1
 
-UNAME=$(shell uname)
```
Comment (abe-winter) [inline Makefile:7]: I think that without this, the makefile will always fall through to the `else` clause below

(and `GOOS=android make build` fails on my linux laptop with 'not found')

File: decoder.go [hunk 3]
```diff
@@ -59,15 +59,12 @@ func getStreamInfo(url string) string {
 	for i := C.uint(0); i < C.uint(avFormatCtx.nb_streams); i++ {
 		stream := *(**C.AVStream)(unsafe.Pointer(uintptr(unsafe.Pointer(avFormatCtx.streams)) + uintptr(i)*unsafe.Sizeof(*avFormatCtx.streams)))
 		codecParams := stream.codecpar
-
-		codecName := C.GoString(C.avcodec_get_name(codecParams.codec_id))
-
+		// TODO (seanp): Add debug logs for codecs found
+		/// codecName := C.GoString(C.avcodec_get_name(codecParams.codec_id))
```
Comment (abe-winter) [inline decoder.go]: commented old code = intentional?

**Review by abe-winter — APPROVED** [review]
lgtm but left note about build failure on linux laptop

(assign fix to me if you want bc I have a linux env)

**Pull request merged**